nestkernel/node_manager.cpp

```diff
@@ -851,6 +851,38 @@ NodeManager::prepare_nodes()
   LOG( M_INFO, "NodeManager::prepare_nodes", os.str() );
 }
 
+void
+NodeManager::post_run_cleanup()
+{
+#ifdef _OPENMP
+#pragma omp parallel
+  {
+    index t = kernel().vp_manager.get_thread_id();
+#else // clang-format off
+  for ( index t = 0; t < kernel().vp_manager.get_num_threads(); ++t )
+  {
+#endif // clang-format on
+    for ( size_t idx = 0; idx < local_nodes_.size(); ++idx )
+    {
+      Node* node = local_nodes_.get_node_by_index( idx );
```

Comment:
Couldn't you use the `begin()` and `end()` iterators of `local_nodes_` to avoid lookup by index?